### PR TITLE
v1.2 dotnet

### PR DESCRIPTION
**Pull request by laves**

### CODE REVIEW

File: .github/workflows/dotnet-codestyle.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    paths:
+      - '**/*.cs'
+      - '.github/workflows/donet-codestyle.yml'
+
+jobs:
+  check-dotnet-codestyle:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (ErisMik) [inline .github/workflows/dotnet-codestyle.yml:21]: `v2` is being deprecated so now's a good time to update

```suggestion
    - uses: actions/checkout@v3
```

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -40,23 +44,37 @@ jobs:
...
+            test-framework: netcoreapp3.1
+          - dotnet-version: 5.0.x
+            binding-framework: netcoreapp3.0
+            test-framework: net5.0
+          - dotnet-version: 6.0.x
+            binding-framework: net6.0
+            test-framework: net6.0
 
     steps:
     - uses: actions/checkout@v2
```
Comment (ErisMik) [inline .github/workflows/dotnet.yml:66]:
```suggestion
    - uses: actions/checkout@v3
```

File: .github/workflows/dotnet.yml [hunk 3]
```diff
@@ -70,11 +88,8 @@ jobs:
...
```
Comment (ErisMik) [inline .github/workflows/dotnet.yml:89]:
```suggestion
    - uses: actions/checkout@v3
```

**Review by ErisMik — APPROVED** [review]

File: binding/dotnet/Leopard/Leopard.cs [hunk 4]
```diff
@@ -48,20 +48,27 @@ public class Leopard : IDisposable
 
         static Leopard()
         {
-#if NETCOREAPP3_1_OR_GREATER
+
+#if NETCOREAPP3_0_OR_GREATER
```
Comment (mrrostam) [inline binding/dotnet/Leopard/Leopard.cs:52]: Do you know why the min version of .Net core was 3.1 instead of 3.0 before?

Comment (laves) [reply binding/dotnet/Leopard/Leopard.cs:52]: It was a mistake - DllImportResolver has always been supported in Core 3.0+

Comment (mrrostam) [reply binding/dotnet/Leopard/Leopard.cs:52]: We should also reflect this in our docs then

**Pull request merged**